File: 009/Problem9.go
package main

import (
	"fmt"
	"time"
)

func problem() {
	var a, b int
	c, s, found := 0, 1000, false
	for a = 1; a < s/3; a++ {
		for b = a; b < s/2; b++ {
			c = s - a - b

			if a*a+b*b == c*c {
				found = true
				break
			}
		}
		if found {
			break
		}
	}
	fmt.Println(a * b * c)
}

func main() {
	start := time.Now()
	problem()
	elapsed := time.Since(start)
	fmt.Printf("Function took %s", elapsed)
}

File: 009/Problem9.py
import time


def problem():
    c, s, found = 0, 1000, False

    for a in range(1, int(s / 3)):
        for b in range(a, int(s / 2)):
            c = s - a - b
            if a * a + b * b == c * c:
                found = True
                break
        if found:
            break
    print(a * b * c)


start = time.time()
for i in range(1000):
    problem()
end = time.time()
print("Function took " + str(end - start))
